post_info.php

```diff
@@ -320,6 +320,25 @@ function handle_survey_setting($survey_id, $level, $user_id, $action, $data)
     $section_id = null;
     // If the user wants to update a survey
     if ($action == "add_or_update") {
+        // Use preprocessing to decide what to do, to be more precise:
+        //     a. If the survey_choice at the level of the user is null,
+        //        we branch first and then update
+        //     b. If the survey_choice at the level of the user is not null,
+        //        we directly update this survey
+        $decision = determine_action_on_add_or_update($survey_id, $level);
+        if ($decision == "branch_and_update") {
```

Comment:
A `switch` statement would be better here.

Reply:
Sure!